[tanxingzhong] iP

TanXingZhong opened the pull request:
### bobAndSteve

> “Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

bobAndSteve frees your mind of having to remember things you need to do. It's,

- text-based
- easy to learn
- ~~FAST~~ *SUPER FAST* to use

All you need to do is:

1. download it from [here](https://nus-cs2103-ay2425s2.github.io/website/schedule/week4/project.html).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE**!

## Features:

- [x] Managing tasks
- [ ] Managing deadlines (coming soon)
- [ ] Reminders (coming soon)

If you are a Java programmer, you can use it to practice Java too. Here's the `main` method:
```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

TanXingZhong committed:
Rename, Greet, Exit

TanXingZhong committed:
Echo

TanXingZhong committed:
Implemented Add, List

TanXingZhong committed:
Mark as Done

TanXingZhong committed:
ToDos, Events, Deadlines

TanXingZhong committed:
Added Automated Text UI Testing

TanXingZhong committed:
Update README.md

TanXingZhong committed:
Implemented Handle Errors

TanXingZhong committed:
Merge branch 'master' of https://github.com/TanXingZhong/ip

TanXingZhong committed:
Implemented Delete task

TanXingZhong committed:
Updated commands to now use ENUMS

TanXingZhong committed:
Changed task to abstract class

TanXingZhong committed:
Implemented Save

TanXingZhong committed:
Merge branch 'branch-Level-7'

TanXingZhong committed:
Implemented Dates and Times

TanXingZhong committed:
Merge pull request #2 from TanXingZhong/branch-Level-8

Merge with branch-Level-8 Implemented Dates and Times

TanXingZhong committed:
A-MoreOOP

TanXingZhong committed:
Add packages for oop

TanXingZhong committed:
Merge pull request #9 from TanXingZhong/add-gradle-support

Add gradle support

TanXingZhong committed:
Add j unit test

TanXingZhong committed:
Add java doc

TanXingZhong committed:
Add jar to gradle

TanXingZhong committed:
Merge branch 'master' into branch-A-JavaDoc

TanXingZhong committed:
Merge pull request #11 from TanXingZhong/branch-A-JavaDoc

Add java doc

TanXingZhong committed:
Add find feature

TanXingZhong committed:
Merge pull request #12 from TanXingZhong/branch-Level-9

Add find feature

TanXingZhong committed:
Add checkstyle

TanXingZhong committed:
Merge pull request #13 from TanXingZhong/branch-A-CheckStyle

Add checkstyle

TanXingZhong committed:
Add gui

TanXingZhong committed:
Merge pull request #14 from TanXingZhong/branch-Level-10

Add gui

TanXingZhong committed:
Update README.md

adoubleb commented on src/main/java/bobandsteve/command/DeleteCommand.java:
```diff
@@ -0,0 +1,56 @@
+ * Command to delete a task from the task list.
+ * The command expects the user to input the task number to delete.
+ */
+public class DeleteCommand extends Command {
```
Should the class name be changed to a noun instead?

adoubleb commented on src/main/java/bobandsteve/tasklist/TaskList.java:
```diff
@@ -0,0 +1,208 @@
+ * The TaskList is used to manage, load, mark, unmark, delete, find and retrieve tasks.
+ */
+public class TaskList {
+    private final List<Task> taskList = new ArrayList<>();
```
Perhaps, a plural form of task would be better?

adoubleb commented on src/main/java/bobandsteve/tasklist/TaskList.java:
```diff
@@ -0,0 +1,208 @@
+     */
+    @Override
+    public String toString() {
+        StringBuilder sb = new StringBuilder();
```
Maybe it would be more explicit if the variable name be more explicit?

adoubleb commented on src/main/java/bobandsteve/task/Task.java:
```diff
@@ -0,0 +1,72 @@
+public abstract class Task {
+
+    private final String description;
+    private String isDone;
```
Perhaps, the naming might confuse readers that the variable is a boolean instead of a string?

adoubleb reviewed (commented):
Great Job! LGTM. Just a few nits.

TanXingZhong committed:
Rebuild: bobAndSteve jar file

Jar file is outdated due to multiple changes and new features.

Used Gradle shadowjar to build a new jar file.

Gradle is an automation tool to help in the rebuilding process.

TanXingZhong committed:
Refactor command and fix minor nits

TanXingZhong committed:
Add assertion with the purpose for precondition check

This ensures better reliability and helps catch potential issues earlier during development.

Add assertions method in Main, Bobandsteve and eta to verify that returned values are not null.

Assertions were added as a safeguard to maintain consistency across the codebase.

TanXingZhong committed:
Merge pull request #15 from TanXingZhong/branch-A-Assertions

Add assertion with the purpose for precondition check

TanXingZhong committed:
Update code base to ensure coding standard

TanXingZhong committed:
Merge branch 'master' into branch-A-CodeQuality

TanXingZhong committed:
Merge pull request #16 from TanXingZhong/branch-A-CodeQuality

Branch a code quality

aaronlim02 commented on src/main/java/bobandsteve/command/HelpCommand.java:
```diff
@@ -0,0 +1,48 @@
+     * Displays the list of available commands in the application.
+     * This includes descriptions and usage for each command.
+     */
+    public String help() {
+        return "Available Commands:\n" + "-------------------\n"
+                + "HELP      - Displays the list of all commands.\n"
+                + "FIND      - Find tasks base on keyword. Usage: find <keyword>\n"
+                + "MARK      - Marks a task as completed. Usage: mark <task number>\n"
+                + "UNMARK    - Unmarks a completed task. Usage: unmark <task number>\n"
+                + "TODO      - Adds a todo. Usage: todo <task>\n"
+                + "DEADLINE  - Adds a deadline. Usage: deadline <task> /by <date>\n"
+                + "EVENT     - Adds an event. Usage: event <task> /from <start> /to <end>\n"
+                + "LIST      - Displays all tasks.\n"
+                + "DELETE    - Deletes a task. Usage: delete <task number>\n"
+                + "BYE       - Exits the program.\n"
+                + "-------------------\n";
+    }
```
I like that you included this command to guide users!

aaronlim02 commented on src/main/java/bobandsteve/command/MarkCommand.java:
```diff
@@ -0,0 +1,54 @@
+ */
+public class MarkCommand extends Command {
+
+    private int pos = 0;
```
I think it is a good idea to use a more descriptive variable name?

aaronlim02 commented on src/main/java/bobandsteve/parser/Parser.java:
```diff
@@ -0,0 +1,68 @@
+            throw new InvalidCommandException("Invalid command, Enter help to view all commands");
+        }
+
+        // Return the appropriate Command based on the parsed command using if-else
+        if (command == CommandEnum.FIND) {
+            return new FindCommand(input);
+        } else if (command == CommandEnum.BYE) {
+            return new ByeCommand();
+        } else if (command == CommandEnum.HELP) {
+            return new HelpCommand();
+        } else if (command == CommandEnum.LIST) {
+            return new ListCommand();
+        } else if (command == CommandEnum.TODO) {
+            return new TodoCommand(input);
+        } else if (command == CommandEnum.DEADLINE) {
+            return new DeadlineCommand(input);
+        } else if (command == CommandEnum.EVENT) {
+            return new EventCommand(input);
+        } else if (command == CommandEnum.UNMARK) {
+            return new UnmarkCommand(input);
+        } else if (command == CommandEnum.MARK) {
+            return new MarkCommand(input);
+        } else if (command == CommandEnum.DELETE) {
+            return new DeleteCommand(input);
+        } else {
+            throw new InvalidCommandException("Invalid command, Enter help to view all commands");
+        }
```
Maybe using switch instead can make the code more succinct?

aaronlim02 commented on src/main/java/bobandsteve/task/Deadline.java:
```diff
@@ -0,0 +1,47 @@
+    private final LocalDate by;
+    private final LocalTime time;
+
+    /**
+     * Constructs a new Deadline task with the specified description, status, and deadline.
+     * The deadline is represented by a date and time string in the format "YYYY-MM-DD HH:MM".
+     *
+     * @param description The description of the deadline task.
+     * @param isDone The status of the task (completed or not).
+     * @param by The deadline for the task in "YYYY-MM-DD HH:MM" format.
+     * @throws InvalidCommandFormatException If the date and time format is invalid or incorrectly formatted.
+     */
```
Well-written JavaDoc comments to explain your function. Well done!

aaronlim02 reviewed (commented):
Overall a great job!

TanXingZhong committed:
Add sort functionality to sort task by deadline

TanXingZhong committed:
Merge pull request #17 from TanXingZhong/branch-C-Sort

Add sort functionality to sort task by deadline

TanXingZhong committed:
Fix bug in formatDateTime

TanXingZhong committed:
Create gradle.yml

TanXingZhong committed:
Update gradle.yml

TanXingZhong committed:
Refactor find to use stream

TanXingZhong committed:
Add welcome and bye message

TanXingZhong committed:
Update README.md to have user guide

TanXingZhong committed:
Update README.md in docs

TanXingZhong committed:
Update README.md in docs

TanXingZhong committed:
Update chat box UXUI

TanXingZhong committed:
Update chat box UXUI

TanXingZhong committed:
Update chat box UXUI

TanXingZhong committed:
Update chat box UXUI

TanXingZhong committed:
Merge pull request #19 from TanXingZhong/branch-A-BetterGui

Branch a better gui

TanXingZhong committed:
Fix: prevent empty inputs

TanXingZhong committed:
Update README.md

TanXingZhong committed:
Update READM.md

TanXingZhong committed:
Update README.md

TanXingZhong committed:
Update java doc in tasklist find function

TanXingZhong committed:
Update README.md

TanXingZhong committed:
Update assert in main window

TanXingZhong committed:
Update java doc

TanXingZhong committed:
Update find command to give better return result

TanXingZhong committed:
Update list command to return better message

TanXingZhong committed:
Update todo command with better error handling

TanXingZhong committed:
Update find feature
